oraclesrv/utils.py

```diff
@@ -0,0 +1,41 @@
+
+from flask import current_app
+import requests
+
+from oraclesrv.client import client
+
+def get_solr_data(reader, rows=5, sort='entry_date', cutoff_days=5, top_n_reads=10):
+    """
+
+    :param reader:
+    :param rows:
+    :param sort:
+    :param cutoff_days:
+    :param top_n_reads:
+    :return:
+    """
+    query = '(similar(topn({topn}, reader:{reader}, {sort} desc)) entdate:[NOW-{cutoff_days}DAYS TO *])'.format(
+                     topn=top_n_reads, reader=reader, sort=sort, cutoff_days=cutoff_days)
+    try:
+        response = client().get(
+            url=current_app.config['ORACLE_SERVICE_SOLRQUERY_URL'],
+            headers={'Authorization': 'Bearer ' + current_app.config['ORACLE_SERVICE_ADSWS_API_TOKEN']},
```

Comment:
oh yes, the discussion today reminded me of this point; so you'll have to decide between two options:

1. ratelimit the endpoint (which calls get_solr_data) so that users cannot abuse it
2. use the user's token (which gets passed from adsws) to make certain calls (i.e. solr search, obviously, you can't use the same token to access `/session` data) and for the others use service's own token